### PR TITLE
PYIC-5211: use dynamic month for the inset texts

### PR DESCRIPTION
**Pull request by shivanshuit914**
### What changed

- use dynamic month for the inset texts

### Why did it change

- To pass month dynamically to inset text

### Issue tracking

- [PYIC-5211](https://govukverify.atlassian.net/browse/PYIC-5211)

[PYIC-5211]: https://govukverify.atlassian.net/browse/PYIC-5211?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by merlinc — CHANGES_REQUESTED** [review]
I know this is a WIP draft, when this PR is ready, I'd like to see there be specifically coded `it` tests for the various permutations of working with the additional `months` data

File: tests/unit/src/presenters/question-to-inset.test.js [hunk 1]
```diff
@@ -4,42 +4,82 @@ const moment = require("moment");
...
   let question;
-  let locals;
+  let data;
   const language = "en";
 
   beforeEach(() => {
     question = {
       questionKey: "rti-payslip-national-insurance",
+      info: {
+        months: 3,
```
Comment (blakeyp) [inline tests/unit/src/presenters/question-to-inset.test.js:14]: In `questions.json` which mocks up what we'll get from the API, the months value is a string

File: tests/unit/src/presenters/question-to-inset.test.js [hunk 2]
```diff
@@ -4,42 +4,82 @@ const moment = require("moment");
...
+
+    presenters.questionToInset(question, translate, language);
+
+    expect(translate).toHaveBeenCalledWith(
+      "fields.rti-payslip-national-insurance.inset",
+      data
+    );
+  });
+
+  it("should call translate using questionID without month when month is 0", () => {
```
Comment (blakeyp) [inline tests/unit/src/presenters/question-to-inset.test.js:53]: `when month is undefined` ? (we won't expect it to be provided at all if not needed)

File: tests/unit/src/presenters/question-to-inset.test.js [hunk 3]
```diff
@@ -4,42 +4,82 @@ const moment = require("moment");
...
-        .subtract(3, "months")
+        .subtract(question.info.months, "months")
         .locale(language)
         .format("DD MMMM YYYY"),
     };
 
     translate = jest.fn();
   });
 
+  it("should call translate using questionID and months", () => {
```
Comment (blakeyp) [inline tests/unit/src/presenters/question-to-inset.test.js:27]: I don't think we need this case, it's covered by the others

File: src/locales/cy/pages.yml [hunk 4]
```diff
@@ -21,6 +21,7 @@ answer-security-questions:
       <p>Os hoffech wybod mwy am sut y bydd eich manylion yn cael eu defnyddio i brofi eich hunaniaeth,
       <a href='https://signin.account.gov.uk/privacy-notice' class='govuk-link' rel='noreferrer noopener' target='_blank'>
       darllenwch ein hysbysiad preifatrwydd (agor mewn tab newydd)</a>.</p>
+  abandon: "<a href='/kbv/abandon' class='govuk-link' data-id='abandon'>Rwyf angen profi fy hunaniaeth mewn ffordd arall</a>"
```
[the remark this answers is not in the document]
Comment (blakeyp) [reply src/locales/cy/pages.yml:24]: Will confirm with UCD and we can update this back later

File: tests/unit/src/presenters/question-to-inset.test.js [hunk 5]
```diff
@@ -1,45 +1,91 @@
...
+    presenters.questionToInset(question, translate, englishLanguage);
 
     expect(translate).toHaveBeenCalledWith(
       "fields.rti-payslip-national-insurance.inset",
-      locals
+      {}
+    );
+  });
+
+  it("should retrun empty string when translation not present", () => {
```
Comment (blakeyp) [inline tests/unit/src/presenters/question-to-inset.test.js:36]: typo *return

File: tests/unit/src/presenters/question-to-inset.test.js [hunk 6]
```diff
@@ -1,45 +1,91 @@
...
 describe("question-to-inset", () => {
   let translate;
   let question;
-  let locals;
-  const language = "en";
+  let data;
+  const englishLanguage = "en";
+  const welshLanguage = "cy";
+  const welshDate = "01 Chwefror 2024";
+  const englishDate = "01 February 2024";
```
Comment (blakeyp) [inline tests/unit/src/presenters/question-to-inset.test.js:12]: UCD will want it in the form `1 February 2024` (no leading 0s) so I think the moment format will need updating to `D MMMM YYYY` ?

**Review by blakeyp — APPROVED** [review]

**Review by merlinc — APPROVED** [review]

**Pull request merged**